worlds/ladx/ItemIconGuessing.py

```diff
@@ -0,0 +1,197 @@
+FORBIDDEN = [
+    "TRADING",
+    "ITEM",
+    "BAD",
+    "SINGLE",
+    "UPGRADE",
+    "BLUE",
+    "RED",
+    "NOTHING",
+    "MESSAGE",
+    "POWER",
+    "MAGIC",
+    "PEGASUS",
+    "TAIL",
+    "ANGLER",
+    "FACE",
+    "BIRD",
+    "SLIME",
+    "GOLD",
+    "MAX",
+    "UPGRADE",
+    "BOMBS",
+    "ARROWS",
+    "NIGHTMARE",
+    "STONE",
+]
+
+SYNONYMS = {
+  'ANKLET': 'POWER_BRACELET',
+  'ARMLET': 'POWER_BRACELET',
+  'BAND': 'POWER_BRACELET',
+  'BANGLE': 'POWER_BRACELET',
+  'BRACER': 'POWER_BRACELET',
+  'CARRY': 'POWER_BRACELET',
+  'CIRCLET': 'POWER_BRACELET',
+  'CROISSANT': 'POWER_BRACELET',
+  'GAUNTLET': 'POWER_BRACELET',
+  'GLOVE': 'POWER_BRACELET',
+  'RING': 'POWER_BRACELET',
+  'STRENGTH': 'POWER_BRACELET',
+  'AEGIS': 'SHIELD',
+  'AEGISSHIELD': 'SHIELD', # Final Fantasy 1
+  'BUCKLER': 'SHIELD',
+  'SHLD': 'SHIELD',
+  'BALLISTA': 'BOW',
+  'GRAPPLE': 'HOOKSHOT',
+  'GRAPPLING': 'HOOKSHOT',
+  'ROPE': 'HOOKSHOT',
+  'BEAM': 'MAGIC_ROD',
+  'CANE': 'MAGIC_ROD',
+  'STAFF': 'MAGIC_ROD',
+  'WAND': 'MAGIC_ROD',
+  'BOOT': 'PEGASUS_BOOTS',
+  'GREAVES': 'PEGASUS_BOOTS',
+  'RUN': 'PEGASUS_BOOTS',
+  'SHOE': 'PEGASUS_BOOTS',
+  'SHOES': 'PEGASUS_BOOTS',
+  'SPEED': 'PEGASUS_BOOTS',
+  'FLUTE': 'OCARINA',
+  'RECORDER': 'OCARINA',
+  'JUMP': 'FEATHER',
+  'PLUME': 'FEATHER',
+  'WINGS': 'FEATHER',
+  'DIG': 'SHOVEL',
+  'BAG': 'MAGIC_POWDER',
+  'CASE': 'MAGIC_POWDER',
+  'DUST': 'MAGIC_POWDER',
+  'POUCH': 'MAGIC_POWDER',
+  'SACK': 'MAGIC_POWDER',
+  'BOMBS': 'BOMB',
+  'BOMBCHUS': 'BOMB',
+  'TNT': 'BOMB',
+  'BLADE': 'SWORD',
+  'CUT': 'SWORD',
+  'DAGGER': 'SWORD',
+  'DIRK': 'SWORD',
+  'EDGE': 'SWORD',
+  'EPEE': 'SWORD',
+  'EXCALIBUR': 'SWORD',
+  'FALCHION': 'SWORD',
+  'KATANA': 'SWORD',
+  'KNIFE': 'SWORD',
+  'MACHETE': 'SWORD',
+  'MASAMUNE': 'SWORD',
+  'MURASAME': 'SWORD',
+  'SABER': 'SWORD',
+  'SABRE': 'SWORD',
+  'SCIMITAR': 'SWORD',
+  'SLASH': 'SWORD',
+  'VORPAL': 'SWORD', # Final Fantasy 1
+  'XCALBER': 'SWORD', # Final Fantasy 1
+  'FLIPPER': 'FLIPPERS',
+  'SWIM': 'FLIPPERS',
+  'BOTTLE': 'MEDICINE',
+  'FLASK': 'MEDICINE',
+  'LEMONADE': 'MEDICINE',
+  'OXYALE': 'MEDICINE', # Final Fantasy 1
+  'POTION': 'MEDICINE',
+  'TEA': 'MEDICINE',
+  'HERB': 'GOLD_LEAF',
+  'CARAPACE': 'SEASHELL',
+  'CONCH': 'SEASHELL',
+  'SHELL': 'SEASHELL',
+  'BOOMER': 'BOOMERANG',
+  'BEAST': 'BOWWOW',
+  'PET': 'BOWWOW',
+  'ARROWS': 'SINGLE_ARROW',
+  'MISSILE': 'SINGLE_ARROW',
+  'QUIVER': 'SINGLE_ARROW',
+  'BIRD': 'ROOSTER',
+  'CHICKEN': 'ROOSTER',
+  'COJIRO': 'ROOSTER', # Ocarina of Time
+  'CUCCO': 'ROOSTER',
+  'FLY': 'ROOSTER',
+  'GRIFFIN': 'ROOSTER',
+  'GRYPHON': 'ROOSTER',
+  'ARMOR': 'BLUE_TUNIC',
+  'DRAGONARMOR': 'BLUE_TUNIC', # Final Fantasy 1
+  'MAIL': 'BLUE_TUNIC',
+  'SUIT': 'BLUE_TUNIC',
+  'TANK': 'HEART_CONTAINER',
+  'FUNGAL': 'TOADSTOOL',
+  'FUNGUS': 'TOADSTOOL',
+  'MUSHROOM': 'TOADSTOOL',
+  'SHROOM': 'TOADSTOOL',
+  'KEY': 'KEY',
```

Comment:
Ideally the system should be able to pick up on identical named items

Reply:
Those were in the original so I left them in. I figured it might be to make sure a different type of key doesn't get used instead of the small key. I'll remove it and test.

Reply:
![P3_ladx_qBWLCehtTk651R1kccoSDQ-230921-145220](https://github.com/ArchipelagoMW/Archipelago/assets/26460970/e6211e2f-ad8c-410f-b0a4-c7db04bc4d83)

This happens with `KEY:KEY` removed. Should be able to safely remove `MAP:MAP` and `COMPASS:COMPASS` though.

Alternatively there's an argument for nightmare keys being the default key icon, and instead doing a specific check for 'SMALL KEY' (like we currently do for 'BIG KEY')

Reply:
Arggh. I knew that was there for a reason. I like the small keys being default, put it back with a comment. My bad for not leaving a comment before.